Fix typo in ProjectRequest

jboxman opened the pull request:
This fixes a typo in ProjectRequest.

So I think this updates fixes this here:

https://docs.openshift.com/container-platform/4.6/rest_api/project_apis/projectrequest-project-openshift-io-v1.html

soltysh reviewed (approved):
/lgtm
/approve

soltysh commented:
@jboxman you need to run `make update` to make this valid PR.

jboxman commented:
@soltysh, cool, I'll sort out the prerequisites for that command to run successfully locally and update.

jboxman commented:
/retest

jboxman commented:
@soltysh, I'm lost on what the prow failure means. I've never attempted this before.

soltysh commented:
> @soltysh, I'm lost on what the prow failure means. I've never attempted this before.

@jboxman in your particular case if you look at the failure from our CI (https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_api/814/pull-ci-openshift-api-master-verify/1338577723466977280) you'll notice that you're missing generated changes to `project/v1/generated.proto`. Ideally you should be able to run `make update` and that should update this file, but if not try just running `hack/update-protobuf.sh` which is required in your case. For it to succeed you'll need protoc which you can get from https://github.com/protocolbuffers/protobuf/releases/tag/v3.0.2 you need to pick one of the `protoc-3.0.2-*` which matches your system.

jboxman commented:
It seems I was unsuccessful:

```
PATH=$PATH:(echo $GOPATH)/bin make update
Running `gofmt -s -l -w` on 261 file(s).
Using existing controller-gen from "_output/tools/bin/controller-gen"
Using existing yq from "_output/tools/bin/yq"
Using existing yaml-patch from "_output/tools/bin/yaml-patch"
'_output/tools/bin/controller-gen' schemapatch:manifests="./authorization/v1" paths="./authorization/v1" output:dir="./authorization/v1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./config/v1" paths="./config/v1" output:dir="./config/v1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./helm/v1beta1" paths="./helm/v1beta1" output:dir="./helm/v1beta1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./console/v1" paths="./console/v1" output:dir="./console/v1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./imageregistry/v1" paths="./imageregistry/v1" output:dir="./imageregistry/v1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./operator/v1" paths="./operator/v1" output:dir="./operator/v1"
_output/tools/bin/yaml-patch -o './operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml-patch' < './operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml' > './operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml.patched'
mv './operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml.patched' './operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml'
_output/tools/bin/yaml-patch -o './operator/v1/0000_25_kube-controller-manager-operator_01_config.crd.yaml-patch' < './operator/v1/0000_25_kube-controller-manager-operator_01_config.crd.yaml' > './operator/v1/0000_25_kube-controller-manager-operator_01_config.crd.yaml.patched'
mv './operator/v1/0000_25_kube-controller-manager-operator_01_config.crd.yaml.patched' './operator/v1/0000_25_kube-controller-manager-operator_01_config.crd.yaml'
_output/tools/bin/yaml-patch -o './operator/v1/0000_25_kube-scheduler-operator_01_config.crd.yaml-patch' < './operator/v1/0000_25_kube-scheduler-operator_01_config.crd.yaml' > './operator/v1/0000_25_kube-scheduler-operator_01_config.crd.yaml.patched'
mv './operator/v1/0000_25_kube-scheduler-operator_01_config.crd.yaml.patched' './operator/v1/0000_25_kube-scheduler-operator_01_config.crd.yaml'
_output/tools/bin/yaml-patch -o './operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml-patch' < './operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml' > './operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml.patched'
mv './operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml.patched' './operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml'
_output/tools/bin/yaml-patch -o './operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml-patch' < './operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml' > './operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml.patched'
mv './operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml.patched' './operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml'
'_output/tools/bin/controller-gen' schemapatch:manifests="./operator/v1alpha1" paths="./operator/v1alpha1" output:dir="./operator/v1alpha1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./operatoringress/v1" paths="./operatoringress/v1" output:dir="./operatoringress/v1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./quota/v1" paths="./quota/v1" output:dir="./quota/v1"
_output/tools/bin/yq m -i -x './quota/v1/0000_03_quota-openshift_01_clusterresourcequota.crd.yaml' './quota/v1/0000_03_quota-openshift_01_clusterresourcequota.crd.yaml-merge-patch'
'_output/tools/bin/controller-gen' schemapatch:manifests="./samples/v1" paths="./samples/v1" output:dir="./samples/v1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./security/v1" paths="./security/v1" output:dir="./security/v1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./securityinternal/v1" paths="./securityinternal/v1" output:dir="./securityinternal/v1"
_output/tools/bin/yaml-patch -o './securityinternal/v1/0000_03_securityinternal-openshift_02_rangeallocation.crd.yaml-patch' < './securityinternal/v1/0000_03_securityinternal-openshift_02_rangeallocation.crd.yaml' > './securityinternal/v1/0000_03_securityinternal-openshift_02_rangeallocation.crd.yaml.patched'
mv './securityinternal/v1/0000_03_securityinternal-openshift_02_rangeallocation.crd.yaml.patched' './securityinternal/v1/0000_03_securityinternal-openshift_02_rangeallocation.crd.yaml'
'_output/tools/bin/controller-gen' schemapatch:manifests="./network/v1" paths="./network/v1" output:dir="./network/v1"
'_output/tools/bin/controller-gen' schemapatch:manifests="./operatorcontrolplane/v1alpha1" paths="./operatorcontrolplane/v1alpha1" output:dir="./operatorcontrolplane/v1alpha1"
hack/update-deepcopy.sh
Generating deepcopy funcs
hack/update-protobuf.sh
topological order bytes
topological order encoding/base64
topological order encoding/json
topological order errors
topological order fmt
topological order github.com/gogo/protobuf/proto
topological order github.com/gogo/protobuf/sortkeys
topological order github.com/google/gofuzz
topological order github.com/openshift/api/image/docker10
topological order github.com/openshift/api/image/dockerpre012
topological order github.com/openshift/api/operator/v1
topological order github.com/openshift/api/pkg/serialization
topological order go/ast
topological order go/doc
topological order go/parser
topological order go/token
topological order gopkg.in/inf.v0
topological order io
topological order k8s.io/api/extensions/v1beta1
topological order k8s.io/apimachinery/pkg/apis/meta/v1/unstructured
topological order k8s.io/apimachinery/pkg/conversion
topological order k8s.io/apimachinery/pkg/conversion/queryparams
topological order k8s.io/apimachinery/pkg/fields
topological order k8s.io/apimachinery/pkg/labels
topological order k8s.io/apimachinery/pkg/selection
topological order k8s.io/apimachinery/pkg/types
topological order k8s.io/apimachinery/pkg/util/errors
topological order k8s.io/apimachinery/pkg/util/json
topological order k8s.io/apimachinery/pkg/util/naming
topological order k8s.io/apimachinery/pkg/util/runtime
topological order k8s.io/apimachinery/pkg/util/sets
topological order k8s.io/apimachinery/pkg/watch
topological order k8s.io/klog/v2
topological order math
topological order math/big
topological order math/bits
topological order net/url
topological order os
topological order reflect
topological order runtime/debug
topological order sigs.k8s.io/structured-merge-diff/v4/value
topological order strconv
topological order strings
topological order sync
topological order sync/atomic
topological order time
topological order unsafe
topological order k8s.io/apimachinery/pkg/runtime/schema
topological order k8s.io/apimachinery/pkg/util/intstr
topological order k8s.io/apimachinery/pkg/api/resource
topological order k8s.io/apimachinery/pkg/runtime
topological order k8s.io/apimachinery/pkg/apis/meta/v1
topological order github.com/openshift/api/network/v1
topological order k8s.io/api/core/v1
topological order k8s.io/api/rbac/v1
topological order k8s.io/apimachinery/pkg/apis/meta/v1beta1
topological order github.com/openshift/api/apps/v1
topological order github.com/openshift/api/build/v1
topological order github.com/openshift/api/image/v1
topological order github.com/openshift/api/oauth/v1
topological order github.com/openshift/api/project/v1
topological order github.com/openshift/api/quota/v1
topological order github.com/openshift/api/route/v1
topological order github.com/openshift/api/samples/v1
topological order github.com/openshift/api/security/v1
topological order github.com/openshift/api/template/v1
topological order github.com/openshift/api/user/v1
topological order github.com/openshift/api/authorization/v1
2020/12/20 17:48:24 github.com/openshift/api/apps/v1/generated.proto:11:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:25 github.com/openshift/api/authorization/v1/generated.proto:13:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:25 github.com/openshift/api/build/v1/generated.proto:11:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:26 github.com/openshift/api/image/v1/generated.proto:12:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:27 github.com/openshift/api/network/v1/generated.proto:10:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:27 github.com/openshift/api/oauth/v1/generated.proto:10:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:27 github.com/openshift/api/project/v1/generated.proto:11:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:27 github.com/openshift/api/quota/v1/generated.proto:11:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:28 github.com/openshift/api/route/v1/generated.proto:11:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
github.com/openshift/api/route/v1/generated.proto:9:1: warning: Import k8s.io/api/core/v1/generated.proto is unused.
2020/12/20 17:48:28 github.com/openshift/api/samples/v1/generated.proto:12:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
github.com/openshift/api/samples/v1/generated.proto:11:1: warning: Import k8s.io/apimachinery/pkg/runtime/generated.proto is unused.
github.com/openshift/api/samples/v1/generated.proto:9:1: warning: Import k8s.io/api/core/v1/generated.proto is unused.
2020/12/20 17:48:28 github.com/openshift/api/security/v1/generated.proto:11:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:28 github.com/openshift/api/template/v1/generated.proto:12:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
2020/12/20 17:48:29 github.com/openshift/api/user/v1/generated.proto:11:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
hack/update-swagger-docs.sh
Generating swagger type docs for apps/v1 at apps/v1
Generating swagger type docs for authorization/v1 at authorization/v1
Generating swagger type docs for build/v1 at build/v1
Generating swagger type docs for console/v1 at console/v1
Generating swagger type docs for config/v1 at config/v1
Generating swagger type docs for image/v1 at image/v1
Generating swagger type docs for imageregistry/v1 at imageregistry/v1
Generating swagger type docs for kubecontrolplane/v1 at kubecontrolplane/v1
Generating swagger type docs for legacyconfig/v1 at legacyconfig/v1
Generating swagger type docs for network/v1 at network/v1
Generating swagger type docs for oauth/v1 at oauth/v1
Generating swagger type docs for openshiftcontrolplane/v1 at openshiftcontrolplane/v1
Generating swagger type docs for operator/v1 at operator/v1
Generating swagger type docs for operatorcontrolplane/v1alpha1 at operatorcontrolplane/v1alpha1
Generating swagger type docs for operatoringress/v1 at operatoringress/v1
Generating swagger type docs for operator/v1alpha1 at operator/v1alpha1
Generating swagger type docs for project/v1 at project/v1
Generating swagger type docs for quota/v1 at quota/v1
Generating swagger type docs for route/v1 at route/v1
Generating swagger type docs for samples/v1 at samples/v1
Generating swagger type docs for security/v1 at security/v1
Generating swagger type docs for securityinternal/v1 at securityinternal/v1
Generating swagger type docs for servicecertsigner/v1alpha1 at servicecertsigner/v1alpha1
Generating swagger type docs for template/v1 at template/v1
Generating swagger type docs for user/v1 at user/v1
 ~/S/w/api   patch-1     git status
On branch patch-1
Your branch is up to date with 'origin/patch-1'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   go.mod
	modified:   go.sum
	modified:   vendor/golang.org/x/net/http2/server.go
	modified:   vendor/golang.org/x/net/http2/transport.go
	modified:   vendor/golang.org/x/tools/go/internal/gcimporter/gcimporter.go
	modified:   vendor/golang.org/x/tools/go/internal/packagesdriver/sizes.go
	modified:   vendor/golang.org/x/tools/go/packages/external.go
	modified:   vendor/golang.org/x/tools/go/packages/golist.go
	modified:   vendor/golang.org/x/tools/go/packages/golist_overlay.go
	modified:   vendor/golang.org/x/tools/go/packages/packages.go
	modified:   vendor/golang.org/x/tools/imports/forward.go
	modified:   vendor/golang.org/x/tools/internal/event/core/event.go
	modified:   vendor/golang.org/x/tools/internal/gocommand/invoke.go
	modified:   vendor/golang.org/x/tools/internal/gopathwalk/walk.go
	modified:   vendor/golang.org/x/tools/internal/imports/fix.go
	modified:   vendor/golang.org/x/tools/internal/imports/imports.go
	modified:   vendor/golang.org/x/tools/internal/imports/mod.go
	modified:   vendor/golang.org/x/tools/internal/imports/sortimports.go
	modified:   vendor/golang.org/x/tools/internal/imports/zstdlib.go
	modified:   vendor/golang.org/x/tools/internal/packagesinternal/packages.go
	modified:   vendor/golang.org/x/tools/internal/typesinternal/types.go
	modified:   vendor/modules.txt

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	vendor/golang.org/x/tools/internal/gocommand/version.go
	vendor/golang.org/x/tools/internal/typesinternal/errorcode.go
	vendor/golang.org/x/tools/internal/typesinternal/errorcode_string.go
```

None of that looks like what I want it to do?

soltysh commented:
@jboxman have you tried running `make generate-with-container` this will invoke all the generation scripts in a pre-built containers, which should be simpler in your case.

jboxman commented:
@soltysh, interesting, that requires `podman`, which is probably best done on Linux. Does anyone develop on MacOS?

soltysh commented:
> @soltysh, interesting, that requires `podman`, which is probably best done on Linux. Does anyone develop on MacOS?

@damemi can you help @jboxman since I remember you're on macos

damemi commented:
@jboxman the output you showed in https://github.com/openshift/api/pull/814#issuecomment-748682976 is correct, the problem is that running locally on darwin produces different protobuf output than linux, which causes CI to fail

I usually just manually run it in a container (from the openshift/api root):
```
$ docker run -it --rm -v $(pwd):/go/src/github.com/openshift/api/.:Z docker.io/openshift/origin-release:golang-1.15 sh
$ cd ../api
$ make update
```

For the most part, docker and podman are interchangeable. But for changes like this, all you need is a linux container to run it in. Hope that helps!

jboxman commented:
@damemi, awesome, that worked perfect, thanks!

soltysh reviewed (approved):
/lgtm
